common/components/utils/truncate.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+
+class Truncate {
+
+  static stringT(str: string, length: number) {
+    var dots = str.length > length ? "..." : "";
+    return str.substring(0, length) + dots;
```

Comment:
To fulfill the function's implied contract we should include the length of the ellipsis.
`return (str + dots).substring(0, length);`

Reply:
In my testing this truncates off the ellipsis, so I'll need to rethink it. I agree that we should include the length of the ellipsis in the returned string, however.

Reply:
Yeah, now that I think about it, maybe something like this:
`return str.substring(0, length - dots.length) + dots;`